### ISSUE TITLE
Finall tensorflow loss for results reported in paper

### ISSUE THREAD

**Issue by Spandan-Madan** [abisee/pointer-generator#28]
Hi @abisee ,

I'm training the network and the tensorflow loss is going down, but really really slow. Frankly, I have always worked with CNN's and they're much faster to train. I just want to get a sense for what is a good value for training loss at which point I can say that the network has learnt something meaningful. I read the issues and you mentioned your model had 230k iterations to reach the results of the paper, and that you had a learning rate schedule where you changed the size of hyperparameters to speed up initial iterations. I'm going for more of a brute force approach right now as I'm trying to get a proof of concept ready for my project, and not final results. 

I'm currently at tensorflow loss of about 2.2 after 2-3 days of training. It started originally at over 6. What is a good value to stop, or what was your model's value for the results in paper?

Thanks :)

**Comment by abisee** [comment]
Yes, RNNs are very slow to train, especially for long sequences (such as in this project), due to the sequential nature of the recurrent connections. I assume by "brute force", you mean you're *not* using smaller values of `max_enc_steps` and `max_dec_steps`. If you're just looking for a "proof of concept" and slowness is your main problem, I'd advise you to try the smaller values of `max_enc_steps` and `max_dec_steps`.

The best way to gauge the quality of your learned model is to run decode mode and look at the output. You can compare it qualitatively to the output provided in the paper and/or the complete output on the test set linked in the README.

**Comment by Spandan-Madan** [comment]
Thanks for the prompt reply @abisee! 

Yes that's what I meant, I am not using smaller values for max_enc_steps and max_dec_steps for now. I am qualitatively tracking progress by decoding as you suggested. But it would be great to know what your value of tensorflow training loss was when you finally stopped training, if you can somehow find the number in your training log files. Will help me out significantly :)

Thanks a lot :)

**Comment by abisee** [comment]
@Spandan-Madan looking at the smoothed loss curve according to Tensorboard, the training loss was about 2.8 after about 230k iterations, before we turned on coverage.

**Comment by Spandan-Madan** [comment]
Hi @abisee, Thanks a lot! This will be quite helpful. By the way, what do you mean by turning on "coverage"?

**Comment by abisee** [comment]
@Spandan-Madan "coverage" is one of the main ideas of the paper. See also [these flags](https://github.com/abisee/pointer-generator/blob/master/run_summarization.py#L63).

**Comment by Spandan-Madan** [comment]
@abisee Oh Damn I remember now, It's been over a month and a half since I read it so it totally slipped my mind sorry! Thanks a lot for pointing to the code :)

**Comment by acohan** [comment]
@abisee can you let us know what was the final running_avg_loss and validation loss for your model? For me, it seems that validation loss stops decreasing after a while and becomes almost flat while the training loss keeps decreasing (overfitting?). Was wondering if you experienced the same trend.
